File: app/src/main/java/me/gilo/wc/adapter/CouponAdapter.java
package me.gilo.wc.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import me.gilo.wc.R;
import me.gilo.wc.adapter.viewholder.CouponViewHolder;
import me.gilo.woodroid.models.Coupon;

import java.util.List;

public class CouponAdapter extends RecyclerView.Adapter<CouponViewHolder> {
    private List<Coupon> coupons;

    public CouponAdapter(List<Coupon> coupons) {
        this.coupons = coupons;
    }

    @Override
    public CouponViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new CouponViewHolder(parent.getContext(), LayoutInflater.from(parent.getContext()).inflate(R.layout.single_coupon_item, parent, false));
    }

    @Override
    public void onBindViewHolder(CouponViewHolder holder, int position) {
        holder.renderView(coupons.get(position));
    }


    @Override
    public int getItemCount() {
        return coupons.size() == 0 ? 0 : coupons.size();
    }
}

File: app/src/main/java/me/gilo/woodroid/app/adapter/CouponAdapter.kt
package me.gilo.woodroid.app.adapter

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import me.gilo.woodroid.app.R
import me.gilo.woodroid.app.adapter.viewholder.CouponViewHolder
import me.gilo.woodroid.models.Coupon

class CouponAdapter(private val coupons: List<Coupon>) : RecyclerView.Adapter<CouponViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CouponViewHolder {
        return CouponViewHolder(
            parent.context,
            LayoutInflater.from(parent.context).inflate(R.layout.single_coupon_item, parent, false)
        )
    }

    override fun onBindViewHolder(holder: CouponViewHolder, position: Int) {
        holder.renderView(coupons[position])
    }


    override fun getItemCount(): Int {
        return if (coupons.size == 0) 0 else coupons.size
    }
}
